go/chat/mobilepush.go

```diff
@@ -0,0 +1,182 @@
+package chat
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/keybase/client/go/chat/globals"
+
+	"github.com/keybase/client/go/chat/storage"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-codec/codec"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+)
+
+type remoteNotificationSuccessHandler struct{}
+
+func (g *remoteNotificationSuccessHandler) HandlerName() string {
+	return "remote notification success"
+}
+func (g *remoteNotificationSuccessHandler) OnConnect(ctx context.Context, conn *rpc.Connection, cli rpc.GenericClient, srv *rpc.Server) error {
+	return nil
+}
+func (g *remoteNotificationSuccessHandler) OnConnectError(err error, reconnectThrottleDuration time.Duration) {
+}
+func (g *remoteNotificationSuccessHandler) OnDisconnected(ctx context.Context, status rpc.DisconnectStatus) {
+}
+func (g *remoteNotificationSuccessHandler) OnDoCommandError(err error, nextTime time.Duration) {}
+func (g *remoteNotificationSuccessHandler) ShouldRetry(name string, err error) bool {
+	return false
+}
+func (g *remoteNotificationSuccessHandler) ShouldRetryOnConnect(err error) bool {
+	return false
+}
+
+type MobilePush struct {
+	globals.Contextified
+	utils.DebugLabeler
+}
+
+func NewMobilePush(g *globals.Context) *MobilePush {
+	return &MobilePush{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "MobilePush", false),
+	}
+}
+
+func (h *MobilePush) AckNotificationSuccess(ctx context.Context, pushIDs []string) {
+	defer h.Trace(ctx, func() error { return nil }, "AckNotificationSuccess")()
+	// Get session token
+	nist, _, err := h.G().ActiveDevice.NISTAndUID(ctx)
```

Comment:
can this get conn/nist situation get wrapped up in a helper? didn't we need this in the extension as well?

Reply:
Sure